pkg/kudoctl/util/repo/package_test.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	frameworkV0FileName = "-framework.yaml"
+	versionV0FileName   = "-frameworkversion.yaml"
+	instanceV0FileName  = "-instance.yaml"
+)
+
+func TestReadFileSystemPackage(t *testing.T) {
+	tests := []struct {
+		name               string
+		v1PackageFolder    string
+		outputGoldenFolder string
+	}{
+		{"zookeeper", "testdata/zk", "testdata/zk-crd-golden"},
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("%s-from-filesystem", tt.name), func(t *testing.T) {
+			actual, err := ReadFileSystemPackage(tt.v1PackageFolder)
+			if err != nil {
+				t.Fatalf("Found unexpected error: %v", err)
+			}
+			golden, err := loadCrdsFromPath(tt.outputGoldenFolder)
+			if err != nil {
+				t.Fatalf("Found unexpected error when loading golden files: %v", err)
+			}
+			// we need to sort here because current yaml parsing is not preserving the order of fields
+			// at the same time, the deep library we use for equality does not support ignoring order
+			sort.Slice(actual.FrameworkVersion.Spec.Parameters, func(i, j int) bool {
+				return actual.FrameworkVersion.Spec.Parameters[i].Name < actual.FrameworkVersion.Spec.Parameters[j].Name
+			})
+			sort.Slice(golden.FrameworkVersion.Spec.Parameters, func(i, j int) bool {
+				return golden.FrameworkVersion.Spec.Parameters[i].Name < golden.FrameworkVersion.Spec.Parameters[j].Name
+			})
+
+			if diff := deep.Equal(golden, actual); diff != nil {
+				t.Error(diff)
+			}
+		})
+
+		t.Run(fmt.Sprintf("%s-from-tarball", tt.name), func(t *testing.T) {
+			appFS := afero.NewMemMapFs()
+			file, err := appFS.Create("testtarball.tar.gz")
+			if err != nil {
+				t.Fatalf("cannot create file for tarball serialization: %+v", err)
+			}
+			defer file.Close()
+			err = createTarball(file, tt.v1PackageFolder)
+			if err != nil {
+				t.Fatalf("cannot create tarball: %+v", err)
+			}
+			file, err = appFS.Open("testtarball.tar.gz")
+			if err != nil {
+				t.Fatalf("could not re-open tarball: %+v", err)
+			}
+			actual, err := ReadTarballPackage(file)
+			if err != nil {
+				t.Fatalf("Found unexpected error: %v", err)
+			}
+			golden, err := loadCrdsFromPath(tt.outputGoldenFolder)
+			if err != nil {
+				t.Fatalf("Found unexpected error when loading golden files: %v", err)
+			}
+			// we need to sort here because current yaml parsing is not preserving the order of fields
+			// at the same time, the deep library we use for equality does not support ignoring order
+			sort.Slice(actual.FrameworkVersion.Spec.Parameters, func(i, j int) bool {
+				return actual.FrameworkVersion.Spec.Parameters[i].Name < actual.FrameworkVersion.Spec.Parameters[j].Name
+			})
+			sort.Slice(golden.FrameworkVersion.Spec.Parameters, func(i, j int) bool {
+				return golden.FrameworkVersion.Spec.Parameters[i].Name < golden.FrameworkVersion.Spec.Parameters[j].Name
+			})
+
+			if diff := deep.Equal(golden, actual); diff != nil {
```

Comment:
I already brought in the `"github.com/stretchr/testify/assert"` package. I think:
```suggestion
			asser.Equal(golden, actual)
```
will work (and print the difference)

Reply:
As discussed in slack. I like that deep does structural diff rather than line based diff, so you end up with

`package_test.go:58: [Framework.Spec.URL: https://zookeeper.apache.org/ != https://zookeeper.apache.org/aaa]` which I found easy to debug